assets/js/modules/subscribe-with-google/components/common/PublicationIDInput.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Subscribe with Google Account Create component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { TextField, Input } from '../../../../material-components';
+import { isValidPublicationID } from '../../util/validation';
+const { useDispatch, useSelect } = Data;
+
+export default function PublicationIDInput() {
+	// Get value.
+	const publicationID = useSelect( ( select ) =>
+		select( STORE_NAME ).getPublicationID()
+	);
+
+	// Handle form input.
+	const { setPublicationID } = useDispatch( STORE_NAME );
+	const onChange = useCallback(
+		( { currentTarget } ) => {
+			setPublicationID( currentTarget.value );
+		},
+		[ setPublicationID ]
+	);
+
```

Comment:
See above comment on `ProductsInput`, technically it would be best to return `null` here if `publicationID` is `undefined` (being loaded). While here it doesn't cause any error currently, it's still the best practice to do so.

Such data being `undefined` is typically only relevant in the very first render cycle, so it will go unnoticed by the user - but without considering `undefined` as a possible value it can cause errors.